doc/lua_api.txt

```diff
@@ -4239,6 +4245,32 @@ Definition tables
         ^ Shall drop item and return the leftover itemstack
         ^ The dropper may be any ObjectRef or nil.
         ^ default: minetest.item_drop ]]
+        on_pickup = func(item_luaentity, picker, time_from_last_punch, tool_capabilities, dir),
+        --[[
+        ^ Called when a dropped item is punched and picked up by `picker`.
+        ^ Function should remove dropped item and add itemstack to `picker`'s
+          inventory, or return `true`.
+        ^ `item_luaentity` is the luaentity of the dropped item.
+        ^ The `picker` may be any `ObjectRef` or `nil`.
+        ^ Not setting this function or returning `true` will first call all functions
```

Comment:
This function is called before running register_on_item_pickup callbacks, and returning false will cancel their execution